excel_import_export/controllers/report.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2019 Ecosoft Co., Ltd (http://ecosoft.co.th/)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+
+import base64
+import json
+import logging
+
+from werkzeug.urls import url_decode
+
+from odoo import http
+from odoo.http import content_disposition, request, route, serialize_exception
+from odoo.tools import html_escape
+from odoo.tools.safe_eval import safe_eval, time
+
+from odoo.addons.web.controllers import report
+
+_logger = logging.getLogger(__name__)
+
+
+class ReportController(report.ReportController):
+    @route()
+    def report_routes(self, reportname, docids=None, converter=None, **data):
+        if converter == "excel":
+            report = request.env["ir.actions.report"]._get_report_from_name(reportname)
+            context = dict(request.env.context)
+            if docids:
+                docids = [int(i) for i in docids.split(",")]
+            if data.get("options"):
+                data.update(json.loads(data.pop("options")))
+            if data.get("context"):
+                # Ignore 'lang' here, because the context in data is the one
+                # from the webclient *but* if the user explicitely wants to
+                # change the lang, this mechanism overwrites it.
+                data["context"] = json.loads(data["context"])
+                if data["context"].get("lang"):
+                    del data["context"]["lang"]
+                context.update(data["context"])
+
+            excel, report_name = report.with_context(**context)._render_excel(
+                docids, data=data
+            )
+            excel = base64.decodebytes(excel)
+            if docids:
+                records = request.env[report.model].browse(docids)
+                if report.print_report_name and not len(records) > 1:
+                    report_name = safe_eval(
+                        report.print_report_name, {"object": records, "time": time}
+                    )
+                    # this is a bad idea, this sould only be .xlsx
+                    extension = report_name.split(".")[-1:].pop()
+                    report_name = f"{report_name}.{extension}"
```

Comment:
@AungKoKoLin1997 
It seems that you took the module while it was still in migration to Odoo 15 (https://github.com/OCA/server-tools/pull/2329).
Before it was fully done, there were some force-pushes that fixed some code. 
Look here: https://github.com/OCA/server-tools/pull/2329/commits/f94ba4fc5c3a15d979424cd043f49d2350e9df60#diff-db4160b3be01859e4c870cbbce8305e536fd4b5cad80ce3812b8e7a187aafccfR51-R56

While migrating, @Mantux11 made an error that switched the lines around. That started generating bad file names. Then he force-pushed to switch them back. 
It looks like you took the code before it was fixed.

TLDR:
```suggestion
                    # this is a bad idea, this should only be .xlsx
                    extension = report_name.split(".")[-1:].pop()
                    report_name = safe_eval(
                        report.print_report_name, {"object": records, "time": time}
                    )
                    report_name = f"{report_name}.{extension}"
```

Reply:
Yes. I worked on these modules while it was still in migration to 15.0. I added the commit for adjustment.